sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcSourceSuite.scala

```diff
@@ -594,4 +594,12 @@ class OrcSourceSuite extends OrcSuite with SharedSparkSession {
     val df = readResourceOrcFile("test-data/TestStringDictionary.testRowIndex.orc")
     assert(df.where("str < 'row 001000'").count() === 1000)
   }
+
+  test("SPARK-33978: Write and read a file with ZSTD compression") {
+    withTempPath { dir =>
+      val path = dir.getAbsolutePath
+      spark.range(3).write.option("compression", "zstd").orc(path)
+      checkAnswer(spark.read.orc(path), Seq(Row(0), Row(1), Row(2)))
```

Comment:
Should we maybe check the explicit compression type? I know one test:

https://github.com/apache/spark/blob/3a299aa6480ac22501512cd0310d31a441d7dfdc/sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcHadoopFsRelationSuite.scala#L85-L108

Reply:
@dongjoon-hyun feel free to skip this comment. Actually, I think the tests are good enough.

Reply:
I avoided it due to the two reasons.
1. `OrcFileOperator` is in `hive` module and this is `sql` module.
2. The ORC native data source file names do not end with `.zstd.orc` postfix.
```
$ ls -al /tmp/zstd/*.orc | head -n1
-rw-r--r--  1 dongjoon  wheel  109 Jan  3 22:41 /tmp/zstd/part-00000-a63d9a17-456f-42d3-87a1-d922112ed28c-c000.orc
```